tasks/update_node_pp.rb

```diff
@@ -9,9 +9,17 @@ def update_file(manifest, target_node)
   _stdout, stderr, status = Open3.capture3("mkdir -p #{path}")
   raise Puppet::Error, _("stderr: ' %{stderr}')" % { stderr: stderr }) if status != 0
   site_path = File.join(path, "#{target_node}.pp")
-  final_manifest = "node '#{target_node}' { #{manifest} } "
+  if File.file?(site_path)
+    existing_manifest = File.readlines(site_path)
+    # remove the last bracket in the manifest
+    existing_manifest.pop
```

Comment:
Is it guaranteed that the manifest will have a braket on the last line?

Reply:
if they are using this task to create / edit. yes it is . 
the path for the manifest updated is unique to the node. controlled by this task. so its a safe assumption.